### DISCUSSION TITLE
Writing Zim Usecases in 'Presentation' Style

### DISCUSSION THREAD

**Discussion by supertoothy** [zim-desktop-wiki/zim-desktop-wiki#2121] [Ideas]
There is an item in the wiki which has been open for a while. See under [Improve Documentation](https://github.com/zim-desktop-wiki/zim-desktop-wiki/wiki/Planning).

` Re-write user manual in a more "presentation" style text, focus on use cases / why you would use something
`

I'd like to start working on this, so I've been thinking about the best way to approach this. My notes on this are as follows.

What I'd like feedback on:

1. Does this rationale for writing usecases make sense?
2. See the example. Is this approach to writing usecases easy to comprehend?
3. What other areas do you think I should write usecases on?

-------------------------------------------------------------------------

### What is this Guide about?

These are Use-case guides. They are different from Walkthroughs or Manuals and serve a specific purpose.

* A Manual documents each feature of the software and it's various facets
* A Walkthrough runs you through the most important features of the software
* A Use Case Guide is from the perspective of the user and their needs

In this sense, in a Use Case guide, the features are introduced in context to user needs and decisions. This is different from both walkthroughs and manuals.

This also makes Use Case Guides opinionated. It is hard to not be prescriptive, when your premise is modeled around a hypothetical user.

Speaking of manuals and walkthroughs, here is a useful list:

* The Zim-Wiki [Manual](https://zim-wiki.org/manual/Start.html) - A Breakdown of features
* Brendan Kidwell's Guide to [Getting Work Done in Zim](https://www.glump.net/content/getting-work-done-in-zim/getting-work-done-in-zim.html) - A Guide to key features in Zim
* Andreas Wehler's [Youtube Playlist](https://www.youtube.com/watch?v=U9bbVFzkN_E&list=PLNY1np4YRcnyl8mWOAih6_wo8G3-OrBpo) - A video walkthrough of key features

### Who is this Guide for?
Use Case guides are for people who prefer to jump-right-in. This assumes that the user interface conventions are straightforward enough that the user can figure things out as they go.

Zim's user-interface is well designed, which makes this approach a viable way to learn how to use Zim. Use case guides are for:

* Users looking for a tool to fit specific workflows
* Users who want to start using Zim right away
* Existing Zim users looking for new perspectives

--------------------------------------------------------------------

I wrote a few usecases to test out the idea:

- Zim at the Office
- Zim as a journalist's research assistant
- Zim helps out at home
- etc

Here is an example:

# 07 A day of Zim at the Office

Jared is the Chief Operating Officer at a fast growing startup. He uses Zim many times a day to manage his daily work needs. Here is what a day in the life of Jared's Zim notebook looks like. 

### Organising the day page
Each morning, Jared grabs a cup of coffee and settles down to see his daily page. Since he has his Journal plugin turned on and this lets him jump to today's page by pressing (alt+d) on his keyboard. Today's page has a list of unstructured line items that Jared had entered the night before, a ritual he has practiced for years.

He uses Zim's formatting capabilities to quickly sort the list, organising them under headings. As the day progresses, these headings become markers that he can jump between using the ToC (Table of Contents plugin) pane.

He also creates a few to-dos for the day, which are also automatically visible in the Task List. (Task List Plugin)

#### Steps in Organising the Day Page

1. Open the Day Page (Alt+d)
4. Formatting text by creating headings (See Format in the menu for text formatting. ctrl+up/down arrow moves lines and paragraphs up or down.)
5. Use the ToC view to jump between headings (You can configure the Table of Contents plugin under Preferences)
6. Create to-dos by typing [] at the start of a line or through the menu via Insert > Checkbox List 

### Attending a Meeting
Jared's first meeting for the day is a daily standup with his team. He always creates a page to record the details for each meeting that he attends. Just before the meeting begins Jared creates a sub-page, right inside his Journal's day page. 

Since Jared uses a pre-designed template for meetings, his new sub-page is already populated with section headers and an expected list of participants. He can immediately start making notes and focus on the discussion.

During the meeting, Jared also opens the page from the last meeting in a seperate window. Zim allows you to open pages in their own window, and move them around. Jared tiles the page from the last meeting alongside the page for the current meeting. This lets him make informed decisions when setting team goals for the week.

When discussing specific projects, Jared jumps to the Project page (ctrl+j) and browses the attachments pane to find a pdf file sent by a vendor, that is related to the discussion.

AT the end of the meeting, Jared uses the *Send to* function to email a copy of his meeting notes to his team.

#### Steps in Attending a Meeting

1. Creating sub-pages within journal pages
2. Creating a meeting page from a template (Key combination: shift+ctrl n or File > New Subpage)
3. Opening pages in a new window (File > Open in a New Window)
4. Navigating using jumps (ctrl+j In the menu Go > Jump to...)  
7. Browsing page attachments (Attachment Browser Plugin)
8. Sending page via email (File > Send to) Opens default mail client

### Creating a Business Proposal
Jared needs to participate in a board meeting later in the day. He wants to create a handout with an outline of his proposal for meeting participants. 

Jared consults his notes from last week, as well as Research Snippets that he had clipped into zim from some websites and whitepapers. He spends a few minutes creating a coherent page structure by moving notes into their right places and renaming them.

The structure of the pages that he has created looks like this:

* Situational Report
	* Summary
	* Hypothesis
		* Hypothesis 1
		* Hypothesis 2
	* Research
		* Source 1
		* Source 2

Now that Jared has a structured source of information, he types out his thesis by adding his notes to each page. When he has completed his writing, he exports the proposal page and it's sub pages as a single-page Markdown document which he can edit and print as meeting handouts.

#### Steps in Creating a Proposal

1. Moving and renaming pages (F2 or File > Rename or move page)
2. Export as Markdown (File > Export...)

**Comment by jaap-karssenberg** [comment]
Hi,

Thanks for taking in interest in improving the documentation.

What I found is that in writing the manual I can describe what a feature
does but often it lacks the context of why you would use it. Also I found
that although the manual is very valuable to technical oriented users who
want to know how a feature works, less technical users don't benefit from
it.

Therefore I was planning to extend the manual with example work flows and
make that the main landing page. This show cases ways to use Zim and makes
less technical users feel more at home. Obviously the manual content would
still be there, cross referenced from the example flows.

A use case in the 3rd person is more intended towards developers. My idea
was more like a description in the 2nd person that puts the features in an
example flow intended for the end user. Sorry for causing confusion but
using the term "use case"in a vague sense.

Regards,

Jaap


Op vr 12 aug. 2022 06:29 schreef gsupreeth ***@***.***>:

**Reply by supertoothy** [reply]
I'm returning to this after almost a year. If this is still on the wishlist, I'd like to give it another shot.

I'm finding it hard to visualise what the end result of what you asked for. Could you write down an example script? Just a paragraph or two would do. I just want to see what a typical end result may look like. Pick any feature that's on your mind, i'm fairly familiar with Zim.

Warm regards,
George
